profile/filter_test.go

```diff
@@ -0,0 +1,404 @@
+		},
+		// Hide
+		{
+			name:       "hide with no matches",
+			hide:       regexp.MustCompile("unknown"),
+			wantMatch:  FilterMatch{fm: true},
+			wantPaths:  allPaths,
+			wantValues: allValues,
+		},
+		{
+			name:      "hide matches function names",
+			hide:      regexp.MustCompile("fun1|fun2"),
+			wantMatch: FilterMatch{fm: true, hm: true},
+			wantPaths: [][]int{
+				{0, 3},
+				{4, 5, 6},
+				{7, 8},
+				{9, 4, 7},
+			},
+			wantValues:      []int64{1, 2, 3, 4},
+			filterLocations: true,
+		},
+		{
+			name:      "hide matches file names",
+			hide:      regexp.MustCompile("file1|file3"),
+			wantMatch: FilterMatch{fm: true, hm: true},
+			wantPaths: [][]int{
+				{0, 2},
+				{4, 5, 6},
+				{7, 8},
+				{9, 4, 7},
+			},
+			wantValues:      []int64{1, 2, 3, 4},
+			filterLocations: true,
+		},
+		{
+			name:      "hide matches mapping names",
+			hide:      regexp.MustCompile("map1"),
+			wantMatch: FilterMatch{fm: true, hm: true},
+			wantPaths: [][]int{
+				{0, 1, 2, 3},
+				{4, 5, 1, 6},
+				{7, 8},
+				{9, 4, 7},
+			},
+			wantValues:      []int64{1, 2, 3, 4},
+			filterLocations: true,
+		},
+		// Compound filters
+		{
+			name:      "hides a stack matched by both focus and ignore",
+			focus:     regexp.MustCompile("fun1|fun7"),
+			ignore:    regexp.MustCompile("fun1"),
+			wantMatch: FilterMatch{fm: true, im: true},
+			wantPaths: [][]int{
+				{7, 8},
+			},
+			wantValues: []int64{3},
+		},
+		{
+			name:      "hides a function if both show and hide match it",
+			show:      regexp.MustCompile("fun1"),
+			hide:      regexp.MustCompile("fun10"),
+			wantMatch: FilterMatch{fm: true, sm: true, hm: true},
+			wantPaths: [][]int{
+				{1},
+				{1},
+			},
+			wantValues:      []int64{1, 2},
+			filterLocations: true,
+		},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			gotProfile := profile.Copy()
+			fm, im, hm, sm := gotProfile.FilterSamplesByName(testCase.focus, testCase.ignore, testCase.hide, testCase.show)
+			gotMatch := FilterMatch{fm: fm, im: im, hm: hm, sm: sm}
+			gotStr := gotProfile.String()
+
+			if gotMatch != testCase.wantMatch {
+				t.Errorf("match got %+v want %+v", gotMatch, testCase.wantMatch)
+			}
+
+			wantSamples, wantLocations := makeFilteredSamples(testCase.wantPaths, testCase.wantValues, testCase.filterLocations, locations)
+			wantProfile := profile.Copy()
+			wantProfile.Sample = wantSamples
+			wantProfile.Location = wantLocations
+			wantStr := wantProfile.String()
+
+			if wantStr != gotStr {
+				diff, err := proftest.Diff([]byte(wantStr), []byte(gotStr))
+				if err != nil {
+					t.Fatalf("failed to get diff: %v", err)
+				} else {
+					t.Errorf("filtered profile got diff(want->got):\n%s", diff)
+				}
+			}
+		})
+	}
+}
+
+// makeFilteredSamples constructs  filtered profile samples and locations from
+// basic inputs to simplify the specification of expected values in filter test
+// cases. paths contains the location indices, indexed by sample and frame, of
```

Comment:
The text is still mostly list of arguments. Could you make it more descriptive in terms of what the function is doing rather than in terms of what arguments are?

Reply:
obsolete